File: LeetCode/DataStruct/Python/Leetcode856-score-of-parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        n,stk = len(s), [0]
        for i in range(n):
            if s[i] == '(': stk.append(0)
            else:
                t = stk.pop()
                if t == 0:
                    stk[-1] += 1
                else:
                    stk[-1] += 2 * t
        return stk[-1]        

File: LeetCode/DataStruct/Golang/Leetcode856-score-of-parentheses.go
package sbgo

func scoreOfParentheses(s string) int {
  n, stk := len(s),make([]int,1)
  for i := 0;i < n;i++ {
    if s[i] == '(' {
      stk = append(stk, 0)
    } else {
      t := stk[len(stk) - 1]
      stk = stk[:len(stk) - 1]
      if t == 0 {
        stk[len(stk) - 1] += 1
      } else {
        stk[len(stk) - 1] += 2 * t
      }
    }
  }
  return stk[0]
}
